app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -137,6 +135,196 @@ object DialogUtils {
             }
         }
     }
+
+    fun colorPickerBottomSheet(
+        context: Context,
+        previousColor: FloatArray,
+        /**
+         * Lambda that provides the string of the current color.
+         * Returns nothing. Used to save the color as the user adjusts it.
+         */
+        onUpdate: ((FloatArray?) -> Unit),
+        /**
+         * Lambda that provides the string of the current color.
+         * Returns nothing. Used to confirm the use of the color.
+         */
+        onSubmit: ((FloatArray?, String?) -> Unit)
+    ) {
+        val sheet = BottomSheetDialog(context)
+        val view = LayoutInflater.from(context).inflate(R.layout.color_dialog, null)
+        val huePicker = view.findViewById<HueView>(R.id.huePicker)
+        val gradientPicker = view.findViewById<GradientView>(R.id.gradientPicker)
+        val hexText = view.findViewById<ForcableEditText>(R.id.hexText)
+        val hueText = view.findViewById<ForcableEditText>(R.id.hueText)
+        val satText = view.findViewById<ForcableEditText>(R.id.satText)
+        val valText = view.findViewById<ForcableEditText>(R.id.valText)
+        val redText = view.findViewById<ForcableEditText>(R.id.redText)
+        val greenText = view.findViewById<ForcableEditText>(R.id.greenText)
+        val blueText = view.findViewById<ForcableEditText>(R.id.blueText)
+        val topColorSquare = view.findViewById<View>(R.id.topColor)
+        val bottomColorSquare = view.findViewById<View>(R.id.bottomColor)
+        val confirmButton = view.findViewById<Button>(R.id.okButton)
+        confirmButton.setOnClickListener {
+            onSubmit.invoke(
+                floatArrayOf(
+                    hueText.floatConvert,
+                    satText.floatConvert / 100,
+                    valText.floatConvert / 100
+                ), hexText.text.toString()
+            )
+            sheet.dismiss()
+        }
+        val cancelButton = view.findViewById<Button>(R.id.cancelButton)
+        cancelButton.setOnClickListener {
+            onSubmit.invoke(null, null)
+            sheet.dismiss()
+        }
+        sheet.setOnCancelListener {
+            onSubmit.invoke(null, null)
+            sheet.dismiss()
+        }
+        /**
+         * Function that updates all the views in the gradient.
+         * Takes in a color state data class to update views.
+         */
+        fun forceViewUpdates(hsvState: ColorUtils.HsvState) {
+            val rgbColor: Int =
+                Color.HSVToColor(floatArrayOf(hsvState.hue, hsvState.saturation, hsvState.value))
+            val hex = java.lang.String.format(
```

Comment:
You're in kotlin, find a Kotlin way to do this without the java namespace. It shouldn't be hard

Reply:
Doing this without the java namespace